apps/system/js/core.js

```diff
@@ -51,16 +52,99 @@
     },
 
     _start: function() {
+      // We need to be sure to get the focus in order to wake up the screen
+      // if the phone goes to sleep before any user interaction.
+      // Apparently it works because no other window
+      // has the focus at this point.
+      window.focus();
+      // With all important event handlers in place, we can now notify
+      // Gecko that we're ready for certain system services to send us
+      // messages (e.g. the radio).
+      // Note that shell.js starts listen for the mozContentEvent event at
+      // mozbrowserloadstart, which sometimes does not happen till
+      // window.onload.
+      this.publish('mozContentEvent', {
+        type: 'system-message-listener-ready'
+      }, true);
+
+      return new Promise((resolve, reject) => {
```

Comment:
This is not the right way to chain promise.

Do you intend for the main chain only resolves when `this.loadWhenIdle()` resolves?
This should be

```javascript
return this.loadWhenIdle([ ... ])
  .then(() => {
    return Promise.all([ ... ]); //L114
  })
  .then(() => {
    // L126, although I am not sure if `developerHud.start();` will return a promise in which we should wait.
  });
```